File: VB/MyXPOClassLibrary/PersistentObject2.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace MyXPOClassLibrary

	Public Class PersistentObject2
		Inherits XPObject
		Public Sub New()
			MyBase.New()
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here.
		End Sub

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here.
		End Sub

		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			' Place here your initialization code.
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
	End Class

End Namespace

File: CS/MyXPOClassLibrary/PersistentObject2.cs
using System;
using DevExpress.Xpo;

namespace MyXPOClassLibrary {

    public class PersistentObject2 : XPObject {
        public PersistentObject2()
            : base() {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here.
        }

        public PersistentObject2(Session session)
            : base(session) {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here.
        }

        public override void AfterConstruction() {
            base.AfterConstruction();
            // Place here your initialization code.
        }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
    }

}